kiwi/bootloader/config/grub2.py

```diff
@@ -227,13 +230,54 @@ def setup_disk_image_config(
                 self.boot_directory_name, 'grub.cfg'
             ]
         )
+
+        if self.validate_use_of_linuxefi:
+            # On systems that uses GRUB_USE_LINUXEFI with grub2 version
+            # less than 2.04 there is no support for dynamic EFI
+            # environment checking. In this condition we extend the grub
+            # setup to add this support. The change kiwi does is as
+            # follows:
+            #
+            # * Apply only on grub < 2.04
+            #    1. Modify 10_linux to set linux/initrd as variables
```

Comment:
This should run after `grub2-mkconfig` got called and only modify `grub.cfg`. There must not be any permanent changes inside `/etc/grub.d/` as they wouldn't be tracked.

Reply:
if we modify the written grub.cfg once and leave 10_linux as is then the next call of grub2-mkconfig will make the $linux variable setup ineffective.

Reply:
Yup, but at that point it's also not necessary anymore as it'll decide correctly between `linux`/`linuxefi` in the running system.